src/net/gateway-bootnode.ts

```diff
@@ -61,34 +64,15 @@ export class GatewayBootnode {
         }
 
         const gw = new Web3Gateway(provider, networkId, environment)
-        return gw.getEnsPublicResolverInstance().then(instance => {
-            let entityEnsNode: string
-            switch (networkId) {
-                case "mainnet":
-                    entityEnsNode = keccak256(VOCDONI_MAINNET_ENTITY_ID)
-                    break
-                case "goerli":
-                    entityEnsNode = keccak256(VOCDONI_GOERLI_ENTITY_ID)
-                    break
-                case "rinkeby":
-                    entityEnsNode = keccak256(VOCDONI_RINKEBY_ENTITY_ID)
-                    break
-                case "xdai":
-                    if (environment === 'prod') {
-                        entityEnsNode = keccak256(VOCDONI_XDAI_ENTITY_ID)
-                        break
-                    }
-                    entityEnsNode = keccak256(VOCDONI_XDAI_STG_ENTITY_ID)
-                    break
-                case "sokol":
-                    entityEnsNode = keccak256(VOCDONI_SOKOL_ENTITY_ID)
-                    break
-            }
-            return instance.text(entityEnsNode, TextRecordKeys.VOCDONI_BOOT_NODES)
-        }).then(uri => {
-            if (!uri) throw new Error("The boot nodes Content URI is not defined on " + networkId)
-            else return new ContentUri(uri)
-        })
+
+        return getEnsPublicResolverByNetwork(gw, { environment, networkId })
```

Comment:
Why not something even more straightforward?

`getEnsTextRecord(ensNode, recordKey, params)`

- No need to have to deal with the contract instance